test/e2e/framework.go

```diff
@@ -600,20 +600,51 @@ func (data *TestData) CreateNamespace(namespace string, mutateFunc func(*corev1.
 		mutateFunc(ns)
 	}
 	if ns, err := data.clientset.CoreV1().Namespaces().Create(context.TODO(), ns, metav1.CreateOptions{}); err != nil {
-		// Ignore error if the namespace already exists
+		// Ignore error if the Namespace already exists
 		if !errors.IsAlreadyExists(err) {
 			return fmt.Errorf("error when creating '%s' Namespace: %v", namespace, err)
 		}
-		// When namespace already exists, check phase
+		// When Namespace already exists, check phase
 		if ns.Status.Phase == corev1.NamespaceTerminating {
 			return fmt.Errorf("error when creating '%s' Namespace: namespace exists but is in 'Terminating' phase", namespace)
 		}
 	}
 	return nil
 }
 
-// createNamespaceWithAnnotations creates the namespace with Annotations.
+func (data *TestData) UpdateNamespace(namespace string, mutateFunc func(*corev1.Namespace)) error {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+	if mutateFunc != nil {
+		mutateFunc(ns)
+	}
```

Comment:
According to the code of `generateNamespaceAnnotationsMutateFunc` and `UpdateNamespace`, this function will erase all labels and annotations that are not provided in the `annotations` parameter of `updateNamespaceWithAnnotations`. I guess it is not intentional.

Should `testAuditLoggingEnableNP` just call the following function directly to avoid making unexpected change?
```
data.UpdateNamespace(namespaces["x"], func(namespace *corev1.Namespace) {
    delete(namespace.Annotations, networkpolicy.EnableNPLoggingAnnotationKey)
})
```
And `UpdateNamespace` should perhaps get the namespace once to avoid overriding other attributes.